osu.Game/Online/Chat/MessageNotifier.cs

```diff
@@ -126,10 +127,15 @@ private void checkForMentions(Channel channel, Message message)
         }
 
         /// <summary>
-        /// Checks if <paramref name="message"/> contains <paramref name="username"/>.
+        /// Checks if <paramref name="message"/> mentions <paramref name="username"/>.
         /// This will match against the case where underscores are used instead of spaces (which is how osu-stable handles usernames with spaces).
         /// </summary>
-        private static bool checkContainsUsername(string message, string username) => message.Contains(username, StringComparison.OrdinalIgnoreCase) || message.Contains(username.Replace(' ', '_'), StringComparison.OrdinalIgnoreCase);
+        public static bool checkContainsUsername(string message, string username)
+        {
+            string fullName = Regex.Escape(username);
+            string underscoreName = Regex.Escape(username.Replace(' ', '_'));
+            return new Regex($"(^|^\\W|\\W|\\w\\W)({fullName}|{underscoreName})($|$\\W|\\W|\\w\\W)", RegexOptions.IgnoreCase).IsMatch(message);
```

Comment:
At this point I think it would rather be easier (and faster) to just split the message and iterate through each word and check in a simple way that would also respect cases like `@username hi` and `username: hi`.

Reply:
So my test are showing that this is indeed more load intensive, but the most time consuming task was still 'only' 1.7ms:

.TestContainsUsernameBetweenInterpunction:
  1) Execution Time: 1115 ticks
  2) Execution Time: 16962 ticks
  
.TestContainsUsernameEndOfLineNegative:
  1) Execution Time: 24 ticks
  2) Execution Time: 2323 ticks
  
.TestContainsUsernameEndOfLinePositive:
  1) Execution Time: 7 ticks
  2) Execution Time: 314 ticks
  
.TestContainsUsernameMidlineNegative:
  1) Execution Time: 5 ticks
  2) Execution Time: 356 ticks

.TestContainsUsernameMidlinePositive:
  1) Execution Time: 6 ticks
  2) Execution Time: 233 ticks

.TestContainsUsernameSpecialCharactersNegative:
  1) Execution Time: 5 ticks
  2) Execution Time: 340 ticks

.TestContainsUsernameSpecialCharactersPositive:
  1) Execution Time: 5 ticks
  2) Execution Time: 253 ticks

.TestContainsUsernameStartOfLineNegative:
  1) Execution Time: 4 ticks
  2) Execution Time: 251 ticks

.TestContainsUsernameStartOfLinePositive:
  1) Execution Time: 5 ticks
  2) Execution Time: 208 ticks

.TestContainsUsernameUnicode:
  1) Execution Time: 170 ticks
  2) Execution Time: 742 ticks

.TestContainsUsernameUnicodeNegative:
  1) Execution Time: 6 ticks
  2) Execution Time: 368 ticks

Reply:
I forgot to mention that I was too lazy to remove the assertion code from the test case which also adds some calculation time to the (2.)-time

Reply:
After simplifying the pattern and taking some time to think about it, I believe that it wouldn't be much faster to split the message into words and check them individually.
Even with splitting, the task of recognizing the username will still be a difficult one for a message like `"@[Tib]: hey mate!"` and the username `"[Tib]"`.
This task will be simplified into checking whether or not the word `"@[Tib]:"` contains the name `"[Tib]"` and couldn't be done with a `word.Contains()` since that wasn't working before.
To correctly detect the username, it would be necessary to get the index of the occurrence in the word and to check, if the characters before and after exist and are not a word-char.
This behavior is exactly what the Regex is doing right now.

I would therefor suggest to keep it this way using a Regex.